### ISSUE TITLE
Accessibility bug at the Details link when the homepage refresh flag is on

### ISSUE THREAD

**Issue by jenniemc** [department-of-veterans-affairs/va.gov-team#21888]
When the Homepage fresh toggle flag is on, the axe scanner detected a critical issue on the appointment summary section. Currently, the `Details` link is tagged with `aria-hidden` however the scanner says focusable elements do not contain aria-hidden.  The expected behavior is the `Details` link should be visible to the screen reader. 

Also, there is a redundant link denoted by the right arrow, `>` going to the appointment details page. 

Solution:
1. Replace the `aria-hidden` label  with `aria-label='Details for appointment on xxx'` on the `Details` link
2. Example of  `aria-label=' can be found in the redundant `<Link></Link>` component immediately following the first
3. Remove the extra `<Link></Link>` component but keep the icon component 
4. Fix all pages that have the `Details` link

final code should look similar such that there's only one `<Link></Link>` component followed by icon component  
      
```
<div>
        <Link
          aria-label={`Details for appointment on ${appointmentDate.format(
            'dddd, MMMM D h:mm a',
          )}`}
          to={isCommunityCare ? `cc/${appointment.id}` : `va/${appointment.id}`}
          className="vads-u-display--none medium-screen:vads-u-display--inline"
        >
          Details
        </Link>
        <i
          aria-hidden="true"
          className="fas fa-chevron-right vads-u-margin-left--1"
        />
</div>
```
